server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamLifecycleWithWarningsTests.java

```diff
@@ -0,0 +1,236 @@
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.IndexSettingProviders;
+import org.elasticsearch.indices.EmptySystemIndices;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.cluster.metadata.DataStreamLifecycleTests.randomDownsampling;
+import static org.elasticsearch.common.settings.Settings.builder;
+import static org.elasticsearch.indices.ShardLimitValidatorTests.createTestShardLimitService;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * We test the warnings added when user configured retention exceeds the global retention in this test,
+ * so we can disable the warning check without impacting all the other test cases
+ */
+public class DataStreamLifecycleWithWarningsTests extends ESTestCase {
+    @Override
+    protected boolean enableWarningsCheck() {
+        // this test expects warnings
+        return false;
+    }
+
+    public void testNoHeaderWarning() {
+        ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
+        HeaderWarning.setThreadContext(threadContext);
+
+        DataStreamLifecycle noRetentionLifecycle = DataStreamLifecycle.newBuilder().downsampling(randomDownsampling()).build();
+        noRetentionLifecycle.addWarningHeaderIfDataRetentionNotEffective(null);
+        Map<String, List<String>> responseHeaders = threadContext.getResponseHeaders();
+        assertThat(responseHeaders.isEmpty(), is(true));
+
+        TimeValue dataStreamRetention = TimeValue.timeValueDays(randomIntBetween(5, 100));
+        DataStreamLifecycle lifecycleWithRetention = DataStreamLifecycle.newBuilder()
+            .dataRetention(dataStreamRetention)
+            .downsampling(randomDownsampling())
+            .build();
+        DataStreamGlobalRetention globalRetention = new DataStreamGlobalRetention(
+            TimeValue.timeValueDays(2),
+            TimeValue.timeValueDays(dataStreamRetention.days() + randomIntBetween(1, 5))
+        );
+        lifecycleWithRetention.addWarningHeaderIfDataRetentionNotEffective(globalRetention);
+    }
+
+    public void testDefaultRetentionHeaderWarning() {
+        ThreadContext threadContext = new ThreadContext(Settings.EMPTY);
+        HeaderWarning.setThreadContext(threadContext);
+
+        DataStreamLifecycle noRetentionLifecycle = DataStreamLifecycle.newBuilder().downsampling(randomDownsampling()).build();
+        DataStreamGlobalRetention globalRetention = new DataStreamGlobalRetention(
+            randomTimeValue(2, 10, TimeUnit.DAYS),
```

Comment:
You are right, and it's not an inconsistency, just trying to keep the PR manageable. `randomTimeValue(2, 10, TimeUnit.DAYS)` was recently introduced, so I use it now in the code I touch. Then in a separate PR I will change it because otherwise we are going to touch many other files.

Reply:
Ah okay that makes sense. I did notice one in this PR [here](https://github.com/elastic/elasticsearch/blob/ffba24b18e78cadcf094bef8a0654fda531d61e1/server/src/test/java/org/elasticsearch/cluster/metadata/DataStreamLifecycleWithWarningsTests.java#L65), that's the only reason I mentioned it :).

Reply:
Yes, you are right!

Reply:
Fixed and added a test to catch it!